airflow/providers/samba/hooks/samba.py

```diff
@@ -36,12 +36,26 @@ def __init__(self, samba_conn_id: str = default_conn_name) -> None:
         self.conn = self.get_connection(samba_conn_id)
 
     def get_conn(self) -> SambaClient:
+        """
+        Return a samba client object.
+
+        You can provide optional parameters in the extra fields of
+        your connection.
+
+        :param logdir: Base directory name for log/debug files.
+        :param kerberos: Try to authenticate with kerberos.
+        :param workgroup: Set the SMB domain of the username.
+        :param netbios_name:
+            This option allows you to override the NetBIOS name that
+            Samba uses for itself.
+        """
         samba = SambaClient(
             server=self.conn.host,
             share=self.conn.schema,
             username=self.conn.login,
             ip=self.conn.host,
             password=self.conn.password,
+            **self.conn.extra_dejson,
```

Comment:
Is the list of fields exhaustive? I feel we should do some sanitising instead of passing the whole JSON object directly in.

Also `:param` is for function arguments and should not be used for `extra` keys. These descriptions should live somewhere else.

Reply:
> By the way, I have since realized that the underlying Python library that supports `SambaHook` is basically abandonware and not particularly useful

Thanks for taising this. It would be worthwhile to open an issue for this to migrate to another library, I think. I don’t have much experience dealing with SMB with Python, but `pysmb` seems to be a popular choice.

Reply:
There is an open issue: https://github.com/apache/airflow/issues/14054

Reply:
@uranusjr so I think we can perhaps go ahead and consider merging this request and then I'll submit another request to migrate to `pysmb`.

I have already done some testing and it seems very well suited.

Reply:
Yup LGTM. Could you close-reopen the PR to trigger CI?

Reply:
@uranusjr done

Reply:
@malthe 
Can you please rebase to latest master?

Reply:
@eladkal done!

Reply:
@eladkal looks about ready now, checks have run.